### PR TITLE
Adds world_to_map() and map_to_world() methods to GridMap and fixes GridMap::set_cell_scale()

### LINKED ISSUE #11856: GridMap enhancement - Convert an arbitrary point in world space, to GridMap cell coordinates
Add a method to GridMap that would convert an arbitrary point to a grid cell coordinate. This is relatively simple to do in GDScript, but it would be a good addition to the base GridMap class.

something like `get_cell_at(Vector3)`

the basic logic for this method would be:
```
var translated_pos = provided_pos / grid_map.get_cell_size()  # convert to cell space
var rounded_pos = Vector3(round(translated_pos.x), round(translated_pos.y), round(translated_pos.z)) # flatten to nearest discrete cell  
return rounded_pos
```
This might need adjustments for coordinates in negative portions along the x, y or z axes, I'm not sure.

### PR DESCRIPTION
**Pull request by dsaltares**

### CODE REVIEW

**Comment by akien-mga** [comment]
How about having an API like TileMap's `world_to_map` and `map_to_world`? https://github.com/godotengine/godot/blob/master/scene/2d/tile_map.cpp#L1310-L1311

**Comment by saltares** [comment]
That makes total sense, it's better for consistency. Will update.

**Comment by saltares** [comment]
@akien-mga did what you suggested and also fixed a bug (see the second commit in this PR).

File: modules/gridmap/grid_map.cpp [hunk 1]
```diff
@@ -333,6 +333,24 @@ int GridMap::get_cell_item_orientation(int p_x, int p_y, int p_z) const {
...
 
+Vector3 GridMap::world_to_map(const Vector3 &p_world_pos) const {
+	Vector3 map_pos = p_world_pos / cell_size;
+	map_pos.x = floor(map_pos.x);
+	map_pos.y = floor(map_pos.y);
+	map_pos.z = floor(map_pos.z);
+	return map_pos;
+}
+
+Vector3 GridMap::map_to_world(int p_x, int p_y, int p_z) const {
```
Comment (saltares) [inline modules/gridmap/grid_map.cpp]: I'm not convinced by this signature myself. In other parts of this file cell coordinates are expressed as 3 separate integers. However, in `world_to_map()` we are forced to return cell coordinates as a `Vector3`. It does feel inconsistent but I understand having 3 separate parameters is the way to enforce cell coordinates are integers....

What do people think?

Comment (akien-mga) [reply modules/gridmap/grid_map.cpp]: I'd say you could use Vector3 to be consistent with the TileMap API.

Then, TileMap also has this mix of `x, y` and `Vector2` methods, with the vector methods with a "v" suffix, like `get_cellv`. Too bad we can't have proper overloading and drop this hack :D
Maybe we should review both APIs before 3.0-beta and make them more consistent, but for the scope of this PR just using Vector3 in map_to_world should be fine.

**Comment by akien-mga** [comment]
Merged manually as 8bc96cc14 with fixed format.

**Pull request closed without merging**